ZenScript/src/main/java/stanhebben/zenscript/expression/ExpressionMapIndexGet.java

```diff
@@ -36,7 +36,7 @@ public ZenType getType() {
     public void compile(boolean result, IEnvironmentMethod environment) {
         if(result) {
             map.compile(result, environment);
-            index.compile(result, environment);
+            index.cast(getPosition(), environment, ((ZenTypeAssociative) map.getType()).getKeyType()).compile(result, environment);
```

Comment:
cast the index before compiling (needed for them to work properly, especially with boxing now)